compiler/resolution/functionResolution.cpp

```diff
@@ -4675,7 +4675,15 @@ static void resolveMaybeSyncSingleField(CallExpr* call) {
     DefExpr*   tmpDefn   = new DefExpr(tmp);
 
     // Applies a type to TMP
-    CallExpr*  tmpExpr   = new CallExpr(PRIM_INIT, fieldDef->exprType->copy());
+    // BHARSH TODO: Why can't we just use the exprType of the formal? Shouldn't
+    // it be the same?
+    Expr*      typeExpr  = fieldDef->exprType->copy();
+    if (ct->isClass()) {
+      // Update symbols
+      update_symbols(typeExpr, &ct->getGenericFieldMap());
```

Comment:
So this is only important in the case of inheritance?  The code you inserted in AggregateType doesn't distinguish between classes and records, unless I missed something

Reply:
I didn't distinguish between classes and records in AgreggateType just to keep things simple and general-purpose, in case we find a use for this map later. I suspect we may need it in other cases, but can't point to anything in particular.